src/OpenRiaServices.Tools/Framework/Logging/CrossProcessLogging.cs

```diff
@@ -98,22 +101,93 @@ void ILogger.LogException(Exception ex)
         var stringBuilder = new StringBuilder();
         string stackTrace = ex?.StackTrace;
 
-        int indentLevel = 0;
-        while (ex != null)
+        FormatException(stringBuilder, ex, indentLevel: 0);
+
+        stringBuilder.AppendLine();
+        stringBuilder.AppendLine($"StackTrace:");
+        stringBuilder.Append('\t');
+        stringBuilder.Append(stackTrace);
+        Write(LogLevel.Error, stringBuilder.ToString());
+    }
+
+    /// <summary>
+    /// Writes <paramref name="ex"/> with inner exceptions and details to <paramref name="stringBuilder"/>
+    /// </summary>
+    /// <remarks>
+    /// The following exception graph
+    /// <code>
+    /// AggregateException
+    /// |-> ArgumentException
+    /// |   |-> ReflectionTypeLoadException
+    /// |        |-> ArgumentNullException 
+    /// |        |-> InvalidArgumentException
+    /// |-> InvalidCastException
+    /// </code>
+    /// Generates an exception similar to 
+    /// <code>
+    ///AggregateException : AGG:message(ICE:message) (AE:message)
+    ///  InvalidCastException : ICE:message
+    ///  ArgumentException : AE:message
+    ///    ReflectionTypeLoadException : RTE:message
+    ///ANE:message(Parameter 'AME:param')
+    ///IOE:message
+    ///    Classes failed to load: 
+    ///     - OpenRiaServices.Tools.Logging.CrossProcessLoggingServer
+    ///     - OpenRiaServices.Tools.Logging.CrossProcessLoggingWriter
+    ///    LoaderExceptions: 
+    ///      ArgumentNullException : ANE:message(Parameter 'AME:param')
+    ///      InvalidOperationException : IOE:message
+    ///StackTrace:
+    ///    at ***
+    /// </code>
+    /// </remarks>
+    static void FormatException(StringBuilder stringBuilder, Exception ex, int indentLevel)
```

Comment:
Consider 
1. moving in callstack and emitting it for aggregate exception /type loader inner
2. Reusing it for "normal" logger